### PR TITLE
Enable ansible lint pre-commit hook

### PR DESCRIPTION
**Pull request by nick-stroud**
This change informs ansible where to look for ansible related files. `pre-commit` now detects `ansible-lint` errors.

```
Finished with 61 failure(s), 6 warning(s) on 312 files.
```

I have not yet done the work of correcting the errors. I played with the `--progressive` option to attempt incremental implementation but could not get it to work with `pre-commit`.

To invoke pre-commit to highlight these errors, without actually committing, run the following command:
```
pre-commit run ansible-lint -a
``` 

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Comment by tpdownes** [comment]
I can't actually make you a reviewer of your own PR, but I believe you should take a look at my updates. I've triggered all 4 full integration tests.

File: tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml [hunk 1]
```diff
@@ -112,16 +120,18 @@
     ## Cleanup and fail gracefully
     rescue:
     - name: Delete Firewall Rule
+      when: fw_result.rc == 0
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml]: I don't like allowing it to always run (and figured out a way of doing it in the HTCondor integration test) but I have made that change.

File: tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml [hunk 2]
```diff
@@ -112,16 +120,18 @@
...
+      failed_when: false
```
[the remark this answers is not in the document]
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml:125]: `ignore_errors: true` is what generates the linting warning. The main difference between the two is that ignore_errors sees the failure and generated all the failure messages, but will keep on going despite the failure. `failed_when` truly overrides the failure.

Arguably, we are in fact willing to ignore an error and `ignore_errors: true` is more correct. What I believe we should be doing is only running the command when we think it should be run (i.e. when we know the firewall rule was created in the first place) but there are a lot of subtleties in getting that right. For now, I'd like to quiet the linter.

I'd consider figuring out how to quiet the linter for these particular `ignore_errors` lines without disabling its ability to warn on the knob (which I think is "bad") generally.

https://medium.com/@sbarnea/why-ansible-ignore-errors-is-evil-500fb6e81229

[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml:125]: SGTM. Sounds like we agree that ignore_error reads better and may be correct but this is the most practical way to move forward.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**